### PR TITLE
Improve interface settings

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:** 

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

---
**What does this PR aim to accomplish?:**

Fix https://github.com/pi-hole/AdminLTE/issues/2009

**How does this PR accomplish the above?:**

Add new option to really bind to the specified interface and take care that the description does not imply we are doing this with the other options. See the referenced issue ticket for why this is confusing.

Accompanying pull requests:

- https://github.com/pi-hole/pi-hole/pull/4476
- https://github.com/pi-hole/docs/pull/620

**What documentation changes (if any) are needed to support this PR?:**

See https://github.com/pi-hole/docs/pull/620

### CODE REVIEW

**Comment by DL6ER** [comment]
Screenshot of the updated settings:

Screenshot from 2021-12-19 10-02-39

**Comment by yubiuser** [comment]
Before we had the warning text only in regard to "all origins". Now we include "respond" (previously "listen") and "bind" into this section. Did our judgement about the danger and firewall necessity change?

**Comment by DL6ER** [comment]
Yes, imagine a user configures a Pi-hole running on a cloud server to listen on the outside interface. Without a firewall this immediately becomes an open resolver.

File: style/pi-hole.css [hunk 1]
```diff
@@ -441,3 +441,17 @@ td.details-control {
...
+  padding-left: 10px;
+  padding-right: 10px;
+}
+
+.danger-area {
+  border-left: 2px solid red;
+  border-right: 2px solid red;
+  padding-left: 10px;
+  padding-right: 10px;
+}
```
Comment (rdwebdesign) [inline style/pi-hole.css]: Suggestion:

```css

.no-danger-area {
  margin: 0.5em 0 1em;
  border-left: 2px solid green;
  border-right: 2px solid green;
  padding: 0.5em 1em;
  box-shadow: 0 0 2px -1px #BBB;
  border-radius: 6px;
}

.danger-area {
  margin: 0.5em 0 1em;
  border-left: 2px solid red;
  border-right: 2px solid red;
  padding: 0.5em 1em;
  box-shadow: 0 0 2px -1px #BBB;
  border-radius: 6px;
}
```
settings_DNS

**Comment by DL6ER** [comment]
Updated screenshot
interface-settings

**Review by yubiuser — APPROVED** [review]

**Pull request merged**